src/type/IntersectionType.php

```diff
@@ -114,13 +114,11 @@ private function ensureNoDuplicateTypes(Type ...$types): void
         foreach ($types as $type) {
             assert($type instanceof ObjectType);
 
-            $names[] = $type->className()->qualifiedName();
-        }
+            $classQualifiedName = $type->className()->qualifiedName();
 
-        if (count(array_unique($names)) < count($names)) {
-            throw new RuntimeException(
-                'An intersection type must not contain duplicate types'
-            );
+            $names[] = in_array($classQualifiedName, $names, true)
```

Comment:
Can we do this without a ternary, please?

Reply:
Sure thing, I will update.